packages/client/src/components/Navigation/Navbar.component.js

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import { MobileNavigation } from '../MobileNavigation/MobileNavigation.component';
+import './Navbar.css';
+
+export const Navbar = ({ isLogedIn, userName, ...props }) => {
+  const [isMobileNavOpen, setIsMobileNavOpen] = useState(false);
+  return (
+    <header className="navigation-bar">
+      <div className="navigation-bar-container">
+        <div className="logo">
+          <Link to="/">
+            <span>KAOSPILOT toolbox</span>
+          </Link>
+        </div>
+        <nav className="navbar-links-container">
+          <Link className="navbar-link" to="/about-toolbox">
+            about toolbox
+          </Link>
+          <Link className="navbar-link" to="/contact-us">
+            contact us
+          </Link>
+
+          <div className={`user-name-container ${!isLogedIn ? 'hide' : ''}`}>
+            <Link className="navbar-link" to="/user-name">
+              <img
+                className="navbar-icon"
+                src="/assets/vectors/vector-person-logged.svg"
+                alt=""
+              />
+              <span> Hi, {userName}!</span>
+            </Link>
+          </div>
+          <div className={`log-in-container ${isLogedIn ? 'hide' : ''}`}>
+            <Link className="navbar-link" to="/log-in">
+              <img
+                className="navbar-icon"
+                src="/assets/vectors/vector-person-not-logged.svg"
+                alt=""
+              />
+              <span>sign in</span>
+            </Link>
+          </div>
+          <div className={`log-out-container ${!isLogedIn ? 'hide' : ''}`}>
+            <Link className="navbar-link" to="/log-out">
+              <img
+                className="navbar-icon"
+                src="/assets/vectors/vector-log-out.svg"
+                alt=""
+              />
+              <span>sign out</span>
+            </Link>
+          </div>
+          <div className="favourite-container">
+            <Link className="navbar-link" to="/favourites">
+              <img
+                className="navbar-icon"
+                src={`${
+                  !isLogedIn
+                    ? '/assets/vectors/vector-favourites-empty-heart.svg'
+                    : '/assets/vectors/vector-favourites-full-heart.svg'
+                }`}
+                alt=""
+              />
+              <span>favourites</span>
+            </Link>
+          </div>
+          <button
+            type="button"
+            className={`mobile-open-btn ${isMobileNavOpen ? 'open' : ''}`}
+            onClick={() => {
+              setIsMobileNavOpen(!isMobileNavOpen);
+            }}
+          >
+            {isMobileNavOpen ? <span>&times;</span> : <span>&#x2630;</span>}
+          </button>
+        </nav>
+        <MobileNavigation open={isMobileNavOpen} />
```

Comment:
```suggestion
        {isMobileNavOpen &&
          <MobileNavigation />
        }
```

In general it's better to a) let parents really control when items are rendered or not (else you just push that code down but have to control the prop value in the parent anyway) and b) only create the DOM elements for what is actually visible.

The less DOM elements are in your page, the better for performance.
If a user is on a desktop, they will never even see the toggle button so why keep spending rendering cycles building this component only to keep it hidden?

Reply:
In this case, we change class-list on MobileComponet using `open `property and we need that to show some transition when this mobile menu appears.

Reply:
You can do animations for when an element is added https://usefulangle.com/post/377/css-animation-on-dom-insert but if you also need to animate when it should be hidden it can become trickier. 
Not a big deal.

Reply:
Yass. I am planning to add a few animations to the design once we have the basics but I promised to freeze everything for some time ;)

Reply:
@miltonhultgren  I've made this way as you suggest and I've made the animation on component rendering (menu opening) but I am totally stuck on how to animate when it is closing because in this case menu just doesn't exist. Maybe I need some extra things for this?

Reply:
Hm maybe something like https://react-spring.dev/hooks/use-transition#usetransition is needed BUT in that case I would consider just sticking with keeping the elements in the DOM for now. 
For the future we could look to "optimize" this IF it's needed.